src/Compilers/Core/Portable/CommandLine/EditorConfig.SectionNameMatching.cs

```diff
@@ -179,7 +316,38 @@ public TokenKind Lex()
             /// <summary>
             /// Call after getting <see cref="TokenKind.SimpleCharacter" /> from <see cref="Lex()" />
             /// </summary>
-            public char EatCurrentCharacter() => _sectionName[_position++];
+            public char EatCurrentCharacter() => _sectionName[Position++];
+
+            public char this[int position] => _sectionName[position];
+
+            /// <summary>
+            /// Returns the string representation of a decimal integer, or null if
+            /// the current lexeme is not an integer.
+            /// </summary>
+            public string TryLexNumber()
+            {
+                bool start = true;
+                var sb = new StringBuilder();
+
+                while (!IsDone)
+                {
+                    char currentChar = EatCurrentCharacter();
+                    if (start && currentChar == '-')
+                    {
+                        sb.Append('-');
```

Comment:
What does a `-` mean in the number representation here?

Reply:
Negative numbers. I really dislike this feature.